module/VuFind/src/VuFind/Db/Service/UserService.php

```diff
@@ -309,16 +337,12 @@ public function getAllUsersWithCatUsernames(): array
      */
     public function getInsecureRows(): array
     {
-        return iterator_to_array($this->getDbTable('User')->getInsecureRows());
-    }
-
-    /**
-     * Create a new user entity.
-     *
-     * @return UserEntityInterface
-     */
-    public function createEntity(): UserEntityInterface
-    {
-        return $this->getDbTable('User')->createRow();
+        $dql = 'SELECT u '
+                . 'FROM ' . $this->getEntityClass(UserEntityInterface::class) . ' u '
+                . 'WHERE u.password != \'\' '
```

Comment:
This might be a little easier to read:

```suggestion
                . "WHERE u.password != '' "
```